server/datastore/mysql/software.go

```diff
@@ -777,6 +800,19 @@ func (ds *Datastore) DeleteSoftwareVulnerabilities(ctx context.Context, vulnerab
 	return nil
 }
 
+func (ds *Datastore) DeleteOutOfDateVulnerabilities(ctx context.Context, source fleet.VulnerabilitySource, duration time.Duration) error {
+	sql := `DELETE FROM software_cve WHERE source = ? AND updated_at < ?`
+
+	var args []interface{}
+	cutPoint := time.Now().UTC().Add(-1 * duration)
+	args = append(args, source, cutPoint)
+
+	if _, err := ds.writer.ExecContext(ctx, sql, args...); err != nil {
+		return ctxerr.Wrapf(ctx, err, "deleting out of date vulnerabilities")
```

Comment:
Nit: `ctxerr.Wrap`

Reply:
Done in https://github.com/fleetdm/fleet/pull/10514/commits/8c2e90168b77aaf402e96e4b18c5e5bc2ce018c8